galsim/real.py

```diff
@@ -457,8 +466,17 @@ def __init__(self, file_name=None, sample=None, image_dir=None, dir=None, preloa
         self.variance = self.cat.field('noise_variance') # noise variance for image
         self.mag = self.cat.field('mag')   # apparent magnitude
         self.band = self.cat.field('band') # bandpass in which apparent mag is measured, e.g., F814W
-        self.weight = self.cat.field('weight') # weight factor to account for size-dependent
-                                               # probability
+        # Add the weight factor to the catalog only if it behaves properly.  In particular, it
+        # should be a float between 0 and 1 (so that random selections of indices can use it to
+        # remove any selection effects in the catalog creation process).  If the weight does not
+        # behave like this, then the catalog simply won't have a weight column.
+        weight = self.cat.field('weight')
+        if np.min(weight) >= 0. and np.max(weight) <= 1. and \
+                not np.any(np.isnan(weight)) and not np.any(np.isinf(weight)):
```

Comment:
The "if" statement is not required for the current catalogs, which were produced with weights that go according to the rules. That's why codecov says we always do the "true" branch.  I could make one that would fail this check for testing purposes, but it sounds like you are suggesting I remove the check, so I will go with that.

We don't currently have separate instructions for producing a catalog.  I tried to make the README sufficiently detailed that it can basically serve as instructions.  The documentation on Sowmya's branch could become a more official set of instructions.